src/components/panel/panel.js

```diff
@@ -1160,3 +1239,131 @@ MdPanelPosition.prototype.right = function(opt_right) {
 MdPanelPosition.prototype.getRight = function() {
   return this._right;
 };
+
+
+
+/*****************************************************************************
+ *                               MdPanelAnimation                            *
+ *****************************************************************************/
+
+
+/**
+ * Animation configuration object. To use, create an MdPanelAnimation with the
+ * desired properties, then pass the object as part of $mdPanel creation.
+ *
+ * Example:
+ *
+ * var panelAnimation = new MdPanelAnimation()
+ *     .openFrom(myButtonEl)
+ *     .closeTo('.my-button')
+ *     .withAnimation('');
+ *
+ * $mdPanel.create({
+ *   animation: panelAnimation
+ * });
+ *
+ * @final @constructor
+ */
+function MdPanelAnimation() {
+  /** @private {!Object} */
+  this._openFrom = {};
+
+  /** @private {!Object} */
+  this._closeTo = {};
+}
+
+
+/**
+ * Gets the boundingClientRect for the opening animation.
+ * @returns {!Object}
+ */
+MdPanelAnimation.prototype.getOpenFrom = function() {
+  return this._openFrom;
+};
+
+
+/**
+ * Gets the boundingClientRect for the closing animation.
+ * @returns {!Object}
+ */
+MdPanelAnimation.prototype.getCloseTo = function() {
+  return this._closeTo;
+};
+
+
+/**
+ * Specifies where to start the open animation. `openFrom` accepts a
+ * click event object, query selector, DOM element, or a Rect object that
+ * is used to determine the bounds. When passed a click event, the location
+ * of the click will be used as the position to start the animation.
+ *
+ * @param {string|!Element|!jQuery.Event|
+ *     {top: string, left: string, height: string, width: string}} openFrom
+ * @returns {MdPanelAnimation}
+ */
+MdPanelAnimation.prototype.openFrom = function(openFrom) {
+  // Check if 'openFrom' is a jQuery.Event.
```

Comment:
ngMaterial doesn't depend on jQuery. Is there anything in the jqlite event that's not on the native event? If not, I think the type can just be `MouseEvent` (since it says it's always click)